libcodechecker/source_code_comment_handler.py

```diff
@@ -0,0 +1,254 @@
+# -------------------------------------------------------------------------
+#                     The CodeChecker Infrastructure
+#   This file is distributed under the University of Illinois Open Source
+#   License. See LICENSE.TXT for details.
+# -------------------------------------------------------------------------
+"""
+Source code comment handling.
+"""
+
+import abc
+import re
+
+from codeCheckerDBAccess_v6 import ttypes
+
+from libcodechecker import util
+from libcodechecker.logger import get_logger
+
+LOG = get_logger('system')
+
+
+class BaseSourceCodeCommentHandler(object):
+    """ Source code handler base class. """
+
+    __metaclass__ = abc.ABCMeta
+
+    __suppressfile = None
+
+    @abc.abstractmethod
+    def store_suppress_bug_id(self,
+                              bug_id,
+                              file_name,
+                              comment,
+                              status):
+        """ Store the suppress bug_id. """
+        pass
+
+    @abc.abstractmethod
+    def remove_suppress_bug_id(self,
+                               bug_id,
+                               file_name):
+        """ Remove the suppress bug_id. """
+        pass
+
+    @property
+    def suppress_file(self):
+        """" File on the filesystem where the suppress
+        data will be written. """
+        return self.__suppressfile
+
+    @suppress_file.setter
+    def suppress_file(self, value):
+        """ Set the suppress file. """
+        self.__suppressfile = value
+
+    @abc.abstractmethod
+    def get_suppressed(self, bug):
+        """
+        Retrieve whether the given bug is suppressed according to the
+        suppress handler.
+        """
+        pass
+
+
+class SourceCodeCommentHandler:
+    """
+    Handle source code comments.
+    """
+    source_code_comment_markers = {
+        'codechecker_suppress': ttypes.ReviewStatus.FALSE_POSITIVE,
+        'codechecker_false_positive': ttypes.ReviewStatus.FALSE_POSITIVE,
+        'codechecker_intentional': ttypes.ReviewStatus.INTENTIONAL,
+        'codechecker_confirmed': ttypes.ReviewStatus.CONFIRMED}
+
+    def __init__(self, source_file):
+        self.__source_file = source_file
+
+    @staticmethod
+    def get_source_code_marker_enum(status):
+        if 'codechecker_' not in status:
+            status = 'codechecker_' + status
+
+        if status not in SourceCodeCommentHandler.source_code_comment_markers:
```

Comment:
I'm not sure we should return `FALSE_POSITIVE` in this case. What if there is a typo `codechecker_confirm` in this case it would return `FALSE_POSITIVE` status?